public-ui/src/app/shared/components/breadcrumb/breadcrumb.component.ts

```diff
@@ -92,31 +99,31 @@ export class BreadcrumbComponent implements OnInit, OnDestroy {
    * Returns array of Breadcrumb objects that represent the breadcrumb trail.
    * Derived from current route in conjunction with the overall route structure.
    */
-  private buildBreadcrumbs(route: ActivatedRoute,
-    url: string = '',
-    breadcrumbs: Breadcrumb[] = []): Array<Breadcrumb> {
+  private buildBreadcrumbs(route: ActivatedRoute, breadcrumbUrl: string = '', breadcrumbs: Breadcrumb[] = []): Array<Breadcrumb> {
     const children: ActivatedRoute[] = route.children;
-    const routeDataBreadcrumb = 'breadcrumb';
     if (children.length === 0) {
       return breadcrumbs;
     }
-    for (const child of children) {
-      if (!child.snapshot.data.hasOwnProperty(routeDataBreadcrumb)) {
-        return this.buildBreadcrumbs(child, url, breadcrumbs);
-      }
-      const routeURL: string = child.snapshot.url.map(segment => segment.path).join('/');
-      if (routeURL.length > 0) {
-        url += `/${routeURL}`;
-      }
-      const label = child.snapshot.data[routeDataBreadcrumb].value;
-      const isIntermediate = child.snapshot.data[routeDataBreadcrumb].intermediate;
+    const [child] = children;
+    const {snapshot: {routeConfig, url}} = child;
```

Comment:
Switched to `routeConfig` instead of `data` since `data` could contain the parent route's data if the current route was empty